jccli/cli/group.py

```diff
@@ -14,29 +20,122 @@ def group(ctx):
 
 @group.command('create')
 @click.option('--name', "-n", required=True, type=str, help='Name of the group')
-@click.option('--type', "-t", required=True, type=click.Choice(['user', 'system']),
-              help='The type of group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
 @click.pass_context
 def create_group(ctx, name, type):
     """
     Create a group
     """
     api2 = JumpcloudApiV2(ctx.obj.get('key'))
-    logger = ctx.obj.get('logger')
+    logger: Logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
     response = api2.create_group(name, type)
-    logger.info(f"{response}")
+    logger.info(f"successfully created group: {name}")
+
+
+@group.command('get')
+@click.option('--name', '-n', required=True, type=str, help='Name of the group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
+@click.pass_context
+def get_group(ctx, name, type):
+    """
+    Detail view of a group
+    """
+    # FIXME: Make the `type` variable required
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    response = api2.get_group(group_name=name, group_type=type)
+    serialized_response = json.dumps(response)
+    logger.info(f"{serialized_response}")
+
+
+@group.command('list')
+@click.option('--user', 'type', flag_value='user_group', help='Restrict to user groups only')
+@click.option('--system', 'type', flag_value='system_group', help='Restrict to system groups only')
+@click.pass_context
+def list_groups(ctx, type):
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger: Logger = ctx.obj.get('logger')
+    response = api2.get_groups(type=type)
+    serialized_response = json.dumps(response, indent=2)
+    logger.info(f"{serialized_response}")
 
 
 @group.command('delete')
 @click.option('--name', "-n", required=True, type=str, help='Name of the group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
 @click.pass_context
-def delete_group(ctx, name):
+def delete_group(ctx, name, type):
     """
     Delete a group
     """
     # FIXME: Ideally, this would output JSON info of the deleted group (similar to delete-user), but at the moment it's
     #  unclear exactly how to do that, given the API wrappers that we have.
     api2 = JumpcloudApiV2(ctx.obj.get('key'))
     logger = ctx.obj.get('logger')
-    api2.delete_group(name)
-    logger.info(f"Group {name} deleted")
+    api2.delete_group(name, type)
```

Comment:
Get `TypeError: 'NoneType' object is not subscriptable` error when attempting to delete a non-existing group.  I also get that error when I attempt to delete a group without specifying a type.  Not sure why we need to pass in a type at all for delete.

➜  jccli group delete -n foo
```
Traceback (most recent call last):
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/bin/jccli", line 8, in <module>
    sys.exit(cli())
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/jccli/cli/group.py", line 84, in delete_group
    api2.delete_group(name, type)
  File "/Users/zaro0508/.virtualenvs/py37-jccli-test1/lib/python3.7/site-packages/jccli/jc_api_v2.py", line 79, in delete_group
    group_id = group['id']
TypeError: 'NoneType' object is not subscriptable```

Reply:
You can have a system group and a user group with the same name. Only a combination of group-name and group-type uniquely identify a group within an org.

Reply:
This is the wrong error for that situation though, which I am fixing right now

Reply:
fixed